### ISSUE TITLE
The OpenApiFilter service does not copy all references from the original document.

### ISSUE THREAD

**Issue by andrueastman** [microsoft/OpenAPI.NET#1799]
Related to https://github.com/microsoft/kiota/issues/5263

Using the OpenApiFilter services does not copy all references in the original document resulting in some missing refs/invalid document.

Taking a look at the visitor at https://github.com/microsoft/OpenAPI.NET/blob/9a5a41521a5650cfecdee799f23266fc5da8cff3/src/Microsoft.OpenApi/Services/CopyReferences.cs#L26

The only references/components copied over are of type schemas,parameters,requestBodies and responses. This leads to leaving out other components that could be referenced like examples/headers/callbacks from the components. See properties at https://github.com/microsoft/OpenAPI.NET/blob/vnext/src/Microsoft.OpenApi/Models/OpenApiComponents.cs#L14

**Comment by maisarissi** [comment]
Hey @andrueastman , will this also fix the issue of having "relative path" in the server URL?
I noticed yesterday by using Kiota that given this OpenAPI description: https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json my sliced OpenAPI description was also changing the server URL to:

```yaml
servers:
  - url: '{protocol}://{hostname}/api/v3'
    variables:
      hostname:
        default: HOSTNAME
        description: Self-hosted Enterprise Server or Enterprise Cloud hostname
      protocol:
        default: http
        description: Self-hosted Enterprise Server or Enterprise Cloud protocol
```

Server URLs for plugin generation always need to be absolute and start with https. So this is making the integration to fail in TTK.

**Comment by MaggieKimani1** [comment]
> Hey @andrueastman @MaggieKimani1 , will this also fix the issue of having "relative path" in the server URL? I noticed yesterday by using Kiota that given this OpenAPI description: https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json my sliced OpenAPI description was also changing the server URL to:
> 
> ```yaml
> servers:
>   - url: '{protocol}://{hostname}/api/v3'
>     variables:
>       hostname:
>         default: HOSTNAME
>         description: Self-hosted Enterprise Server or Enterprise Cloud hostname
>       protocol:
>         default: http
>         description: Self-hosted Enterprise Server or Enterprise Cloud protocol
> ```
> 
> Server URLs for plugin generation always need to be absolute and start with https. So this is making the integration to fail in TTK.

No this seems to be a completely different issue as server information seems to get lost during copying. Let me investigate and work on a separate fix for this.

**Comment by andrueastman** [comment]
Hey @maisarissi,

Any chance you can double check this for us? 

The description at https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/ghes-3.0/ghes-3.0.json
seems to replicate your scenario as the original server object is as you say. 
<img width="842" alt="image" src="https://github.com/user-attachments/assets/cce232fc-89fa-4bf4-8dc1-ae5b8940e9aa">

However, using the document at https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json which you shared does not replicate this scenario.

**Comment by maisarissi** [comment]
So, I used the deep link as the following:
```vscode://ms-graph.kiota/OpenDescription?descriptionUrl%3Dhttps://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json&kind=Plugin&type=ApiPlugin&name=GitHubClientPlugin&source=ttk```

And the sliced OpenAPI description changes the server URL info to:
```yaml
servers:
  - url: '{protocol}://{hostname}/api/v3'
    variables:
      hostname:
        default: HOSTNAME
        description: Self-hosted Enterprise Server or Enterprise Cloud hostname
      protocol:
        default: http
        description: Self-hosted Enterprise Server or Enterprise Cloud protocol
```

**Comment by maisarissi** [comment]
Just figured out the issue.

First time, I ran the deep link using the https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/ghes-3.0/ghes-3.0.json 
Kiota has copied the original OpenAPI description into the `.kiota` folder. But because we were having the issue with the `Ref Examples` and the server URL, I changed to https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json
I guess Kiota was still using the local copy rather then the new URL.
